tests/http_server_test.cc

```diff
@@ -0,0 +1,225 @@
+#include <pistache/http.h>
+#include <pistache/client.h>
+#include <pistache/endpoint.h>
+
+#include "gtest/gtest.h"
+
+#include <chrono>
+#include <future>
+
+using namespace Pistache;
+
+struct HelloHandlerWithDelay : public Http::Handler {
+    HTTP_PROTOTYPE(HelloHandlerWithDelay)
+
+    explicit HelloHandlerWithDelay(int delay = 0) : delay_(delay)
+    { }
+
+    void onRequest(const Http::Request& /*request*/, Http::ResponseWriter writer)
+    {
+        std::this_thread::sleep_for(std::chrono::seconds(delay_));
+        writer.send(Http::Code::Ok, "Hello, World!");
+    }
+
+    int delay_;
+};
+
+constexpr char SPECIAL_PAGE[] = "/specialpage";
+
+struct SlowHandlerOnSpecialPage : public Http::Handler {
+    HTTP_PROTOTYPE(SlowHandlerOnSpecialPage)
+
+    explicit SlowHandlerOnSpecialPage(int delay = 0) : delay_(delay)
+    { }
+
+    void onRequest(const Http::Request& request, Http::ResponseWriter writer)
+    {
+        if (request.resource() == SPECIAL_PAGE)
+        {
+            std::this_thread::sleep_for(std::chrono::seconds(delay_));
+        }
+        writer.send(Http::Code::Ok, "Hello, World!");
+    }
+
+    int delay_;
+};
+
+TEST(http_server_test, client_disconnection_on_timeout_from_single_threaded_server) {
+    const std::string address = "localhost:9095";
```

Comment:
This is fine for now, but we should generate these port numbers dynamically.  Maybe set the port to 0, let the kernel choose it, add an api to pistache (if there is not one already) to ask what port number is in use, and pass that to the client (in the unit test).

These unit tests will fail if the port number is already bound on the test system, which makes these tests non-hermetic.

Reply:
@dennisjenkins75 Yes, you are totally right, I will think about this problem.